crates/sdk/src/network-v2/client.rs

```diff
@@ -0,0 +1,228 @@
+    GetFilteredProofRequestsResponse, GetNonceRequest, GetProofRequestStatusRequest,
+    GetProofRequestStatusResponse, ProofMode, ProofStatus, ProofStrategy, RequestProofRequest,
+    RequestProofRequestBody, RequestProofResponse,
+};
+use crate::network_v2::Signable;
+
+/// The default RPC endpoint for the Succinct prover network.
+pub const DEFAULT_PROVER_NETWORK_RPC: &str = "http://127.0.0.1:50051";
+
+pub struct NetworkClient {
+    signer: PrivateKeySigner,
+    http: HttpClientWithMiddleware,
+    s3: OnceCell<S3Client>,
+}
+
+impl NetworkClient {
+    /// Returns the currently configured RPC endpoint for the Succinct prover network.
+    pub fn rpc_url() -> String {
+        env::var("PROVER_NETWORK_RPC").unwrap_or_else(|_| DEFAULT_PROVER_NETWORK_RPC.to_string())
+    }
+
+    /// Create a new network client with the given private key.
+    pub fn new(private_key: &str) -> Self {
+        let signer = PrivateKeySigner::from_str(private_key).unwrap();
+
+        let http_client = reqwest::Client::builder()
+            .pool_max_idle_per_host(0)
+            .pool_idle_timeout(Duration::from_secs(240))
+            .build()
+            .unwrap();
+
+        Self { signer, http: http_client.into(), s3: OnceCell::new() }
+    }
+
+    /// Get a connected RPC client.
+    async fn get_rpc(&self) -> Result<ProverNetworkClient<Channel>> {
+        let rpc_url = Self::rpc_url();
+        let channel = Channel::from_shared(rpc_url)?.connect().await?;
+        Ok(ProverNetworkClient::new(channel.clone()))
+    }
+
+    /// Get a connected artifact store client.
+    async fn get_store(&self) -> Result<ArtifactStoreClient<Channel>> {
+        let rpc_url = Self::rpc_url();
+        let channel = Channel::from_shared(rpc_url)?.connect().await?;
+        Ok(ArtifactStoreClient::new(channel.clone()))
+    }
+
+    /// Gets the latest nonce for this account's address.
+    pub async fn get_nonce(&self) -> Result<u64> {
+        let mut rpc = self.get_rpc().await?;
+
+        let res =
+            rpc.get_nonce(GetNonceRequest { address: self.signer.address().to_vec() }).await?;
+        Ok(res.into_inner().nonce)
+    }
+
+    /// Get the status of a given proof. If the status is Fulfilled, the proof is also returned.
+    pub async fn get_proof_request_status<P: DeserializeOwned>(
+        &self,
+        request_id: &[u8],
+    ) -> Result<(GetProofRequestStatusResponse, Option<P>)> {
+        let mut rpc = self.get_rpc().await?;
+
+        let res = rpc
+            .get_proof_request_status(GetProofRequestStatusRequest {
+                request_id: request_id.to_vec(),
+            })
+            .await?
+            .into_inner();
+        let status = ProofStatus::try_from(res.status)?;
+        let proof = match status {
+            ProofStatus::Fulfilled => {
+                log::info!("Proof request fulfilled");
+                let proof_uri = res.proof_uri.as_ref().expect("no proof url");
```

Comment:
don't use expect in fn that has Result, return err

Reply:
fixed